tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py

```diff
@@ -0,0 +1,147 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+import requests
+
+try:
+    # py3
+    import urllib.parse as url_parse
+except:
+    # py2
+    import urlparse as url_parse
+
+import subprocess
+
+# the functions we patch
+from azure.core.pipeline.transport import RequestsTransport
+
+# the trimming function to clean up incoming arguments to the test function we are wrapping
+from azure_devtools.scenario_tests.utilities import trim_kwargs_from_test_function
+
+# defaults
+PROXY_URL = "http://localhost:5000"
+RECORDING_START_URL = "{}/record/start".format(PROXY_URL)
+RECORDING_STOP_URL = "{}/record/stop".format(PROXY_URL)
+PLAYBACK_START_URL = "{}/playback/start".format(PROXY_URL)
+PLAYBACK_STOP_URL = "{}/playback/stop".format(PROXY_URL)
+
+# TODO, create a pytest scope="session" implementation that can be added to a fixture such that unit tests can
+# startup/shutdown the local test proxy
+# this should also fire the admin mapping updates, and start/end the session for commiting recording updates
+
+
+IS_LIVE = os.getenv("AZURE_TEST_RUN_LIVE") == "true" or os.getenv("AZURE_TEST_RUN_LIVE") == "yes"
+IS_PLAYBACK = os.getenv("AZURE_TEST_RUN_LIVE") == "false" or os.getenv("AZURE_TEST_RUN_LIVE") == "no"
```

Comment:
Nit: `or not os.getenv("AZURE_TEST_RUN_LIVE")` right?

Reply:
This was following the logic of "skip some functionality if set to neither live nor playback", but this now defaults to playback behavior if `is_live` is false